### PR TITLE
[Merged by Bors] - feat(Topology/Group): drop an unneeded assumption

### PR DESCRIPTION
**Pull request by urkud**
- prove that the quotient map `G → G ⧸ N` is an open quotient map;
- use recently added lemmas to drop a `[LocallyCompactSpace G]` assumption in `QuotientGroup.continuousSMul` (now called `QuotientGroup.instContinuousSMul`).

---

- [x] depends on: #16549

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/Group/Compact.lean [hunk 1]
```diff
@@ -4,24 +4,18 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
-import Mathlib.Topology.CompactOpen
 import Mathlib.Topology.Sets.Compacts
 
 /-!
 # Additional results on topological groups
 
-Two results on topological groups that have been separated out as they require more substantial
-imports developing either positive compacts or the compact open topology.
-
+A result on topological groups that have been separated out
```
Comment (YaelDillies) [inline Mathlib/Topology/Algebra/Group/Compact.lean:12]:
```suggestion
A result on topological groups that has been separated out
```

**Review by YaelDillies — APPROVED** [review]
Would be nice if the PR description could say exactly which hypothesis is removed from what declaration. It's not super clear from the diff.

maintainer merge

**Comment by jcommelin** [comment]
Please update the PR description as @_YaelDillies requested.

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**